staging/src/k8s.io/apimachinery/pkg/util/runtime/runtime.go

```diff
@@ -155,13 +146,17 @@ func GetCaller() string {
 // handlers to handle errors and panics the same way.
 func RecoverFromPanic(err *error) {
 	if r := recover(); r != nil {
-		callers := getCallers(r)
+		// Same as stdlib http server code. Manually allocate stack trace buffer size
+		// to prevent excessively large logs
+		const size = 64 << 10
```

Comment:
I wonder if it would be worth extracting this logic somewhere, since we do the same thing in handlers/rest.go. Probably not though..

Reply:
I agree. I wasn't sure on this

on one hand, I'd imagine extracting this logic would result in one more function call in the call stack

on the other hand, I guess there are more places in our code base need this logic (I was about to add the logic [here](https://github.com/kubernetes/kubernetes/blob/dd96b15e5d1de1923645ef17c05dfdfb616163bf/staging/src/k8s.io/client-go/tools/cache/reflector.go#L179) until I saw https://github.com/kubernetes/kubernetes/pull/73137#discussion_r249467406)